src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -21,11 +21,35 @@ namespace ray {
 namespace gcs {
 
 GcsResourceManager::GcsResourceManager(
+    instrumented_io_context &io_context,
     std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
     ClusterResourceManager &cluster_resource_manager)
-    : gcs_table_storage_(gcs_table_storage),
+    : io_context_(io_context),
+      gcs_table_storage_(gcs_table_storage),
       cluster_resource_manager_(cluster_resource_manager) {}
 
+void GcsResourceManager::Update(std::shared_ptr<const syncer::RaySyncMessage> message) {
+  // Make sure thread safety.
+  io_context_.post(
+      [this, message]() {
+        rpc::ResourcesData resources;
+        resources.ParseFromString(message->sync_message());
```

Comment:
To decrease the load of the main thread, is it better to deserilize the message in current thread?
I am not sure if the copying makes difference between string and message object.

Reply:
@WangTaoTheTonic copy string is faster.

If this is a bottleneck, we'll need to add CPUs thread. The idea here is to make syncer transparent to the message send between nodes for better performance. string is faster than message in terms of copy and serde.